src/tiled/mainwindow.cpp

```diff
@@ -99,6 +102,95 @@ using namespace Tiled::Internal;
 using namespace Tiled::Utils;
 
 
+namespace {
+
+template <typename Format>
+struct ExportDetails
+{
+   Format* mFormat;
+   QString mFileName;
+
+   ExportDetails() {}
+   ExportDetails(Format* format, const QString& fileName)
+     : mFormat(format)
+     , mFileName(fileName)
+   {}
+
+   bool valid() { return static_cast<bool>(mFormat); }
+};
+
+template <typename Format>
+ExportDetails<Format> chooseExportDetails(const QString &fileName,
+                                          const QString &lastExportName,
+                                          const QString &lastExportFilter,
+                                          QWidget* window)
+{
+    FormatHelper<Format> helper(FileFormat::Write, window->tr("All Files (*)"));
```

Comment:
You can't use `window->tr`, since that way `lupdate` is not able to determine the context for the translatable string. Fortunately the function is static, so you can write `MainWindow::tr` instead.

Reply:
Done

Thanks for the tip on that, I was just trying to keep the same calls.